### DISCUSSION TITLE
Pre-create users for external providers

### DISCUSSION THREAD

**Discussion by lukasamd** [zitadel/zitadel#8784] [Q&A]
Hey, we use Zitadel for internal usage and linked it with Google SSO (we use gsuite). Right now onboarding process for new users is a bit slow: they need to start on Zitadel to create account using mentioned Google SSO, after admin need to add proper privileges. We can not resolve that by actions in 100% cases because some users should have higher privileges

So, we would like to create them first when we know they will join, and then link them automatically. Email is the same in both cases

Unfortunately right now it does not work i. e. Users need to register anyway. We do not use emails from Zitadel so it's a bit problematic

Is there any way to make it simpler?

**Comment by hifabienne** [comment]
Hi @lukasamd 
Your use case really makes sense, but we unfortunately don't have it implemented as you wish in our current hosted login.
But we are currently working on our new typescript login, which will replace the current login ui.
This is basically what you are looking for from my understanding: https://github.com/zitadel/typescript/issues/166

In the current login a password is needed for a user, thats why the invite flow for your use case doesn't work as you like.

So basically our new session API, is already capable of the functionality you need. But the current login doesn't use the new session API, thats why we are working on the new login.

You can also build your own login UI on top of the session API if that would be an option for you.

**Comment by lukasamd** [comment]
@hifabienne Hey, thanks for quick and very valuable answer!

Ok, we can modify login UI, but I see it's still a bit experimental repo. Maybe other solution, I need some advice - what if we will pre-create accounts and send verification emails? Then users can follow them, login using Google SSO and link both accounts. They will provide password of course, but we will not use them anyway. Should something like that resolve issue? It will still allow us to resolve two actions from devops/security team - i.e. they will just create account, rest will be on end-user side

**Reply by hifabienne** [reply]
with the current login it is possible that the user adds the password, they can later on use the sso and link it the the existing user.
